src/imports/gallery/qdeclarativedocumentgallery.cpp

```diff
@@ -46,10 +46,12 @@
 #include <QtCore/qmetaobject.h>
 #include <QtQml/qqmlcontext.h>
 #include <QtQml/qqmlengine.h>
+#include <QCoreApplication>
+#include <QDebug>
 
 QT_BEGIN_NAMESPACE_DOCGALLERY
 
-Q_GLOBAL_STATIC(QDocumentGallery, qt_declarativeDocumentGalleryInstance)
+static QDocumentGallery* instance = nullptr;
```

Comment:
Since this is needed / should be accessed only from within that one method -> why not limit visibility by putting the static variable in there?